Lossless video format (using jcodec) not supported by dl4j

I have generated labelled videos using jcodec 1.5 and 1.9, both using the standard SequenceEncoder.

SequenceEncoder encoder = new SequenceEncoder(new File(<Name Video>));
BufferedImage bi = ImageIO.read(<Some Image File>);
encoder.encodeNativeFrame(AWTUtil.fromBufferedImage(bi));
encoder.finish.

This results in lossy videos, see screenshot below.

I also generate videos as was written here: http://stackoverflow.com/questions/11345343/png-image-files-to-video-lossless
 
This results in lossless videos, which are also much smaller (again, see screenshot). However, dl4j does not load these images. In jcodec 1.5 I receive codec not supported, and using jcodec 1.9 I receive the following error :

Exception in thread "main" java.lang.RuntimeException: Exception thrown in base iterator
	at org.deeplearning4j.datasets.iterator.AsyncDataSetIterator.next(AsyncDataSetIterator.java:231)
	at org.deeplearning4j.datasets.iterator.AsyncDataSetIterator.next(AsyncDataSetIterator.java:36)
	at TrackingPredictions.VideoClassification.main(VideoClassification.java:154)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
Caused by: java.lang.NullPointerException
	at org.jcodec.containers.mp4.demuxer.FramesMP4DemuxerTrack.getMeta(FramesMP4DemuxerTrack.java:203)
	at org.jcodec.api.FrameGrab.detectKeyFrame(FrameGrab.java:211)
	at org.jcodec.api.FrameGrab.decodeLeadingFrames(FrameGrab.java:188)
	at org.jcodec.api.FrameGrab.<init>(FrameGrab.java:85)
	at org.datavec.codec.reader.CodecRecordReader.loadData(CodecRecordReader.java:107)
	at org.datavec.codec.reader.CodecRecordReader.sequenceRecord(CodecRecordReader.java:85)
	at org.deeplearning4j.datasets.datavec.SequenceRecordReaderDataSetIterator.nextMultipleSequenceReaders(SequenceRecordReaderDataSetIterator.java:200)
	at org.deeplearning4j.datasets.datavec.SequenceRecordReaderDataSetIterator.next(SequenceRecordReaderDataSetIterator.java:133)
	at org.deeplearning4j.datasets.datavec.SequenceRecordReaderDataSetIterator.next(SequenceRecordReaderDataSetIterator.java:115)
	at org.deeplearning4j.datasets.datavec.SequenceRecordReaderDataSetIterator.next(SequenceRecordReaderDataSetIterator.java:29)
	at org.deeplearning4j.datasets.iterator.AsyncDataSetIterator$IteratorRunnable.run(AsyncDataSetIterator.java:300)
	at java.lang.Thread.run(Thread.java:745)

I assume that lossless videos yield in better learning performances, and that smaller videos most likely in speed performances..

--- a/datavec-data/datavec-data-codec/pom.xml
+++ b/datavec-data/datavec-data-codec/pom.xml
@@ -49,5 +49,50 @@
             <version>${nd4j.version}</version>
             <scope>test</scope>
         </dependency>
+
+<!-- Do not depend on FFmpeg by default due to licensing concerns. -->
+<!--
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>ffmpeg</artifactId>
+            <version>${ffmpeg.version}-${javacpp-presets.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>ffmpeg</artifactId>
+            <version>${ffmpeg.version}-${javacpp-presets.version}</version>
+            <classifier>android-arm</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>ffmpeg</artifactId>
+            <version>${ffmpeg.version}-${javacpp-presets.version}</version>
+            <classifier>android-x86</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>ffmpeg</artifactId>
+            <version>${ffmpeg.version}-${javacpp-presets.version}</version>
+            <classifier>linux-x86_64</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>ffmpeg</artifactId>
+            <version>${ffmpeg.version}-${javacpp-presets.version}</version>
+            <classifier>macosx-x86_64</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>ffmpeg</artifactId>
+            <version>${ffmpeg.version}-${javacpp-presets.version}</version>
+            <classifier>windows-x86_64</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>ffmpeg</artifactId>
+            <version>${ffmpeg.version}-${javacpp-presets.version}</version>
+            <classifier>linux-ppc64le</classifier>
+        </dependency>
+-->
     </dependencies>
 </project>

--- a/datavec-data/datavec-data-codec/src/main/java/org/datavec/codec/reader/BaseCodecRecordReader.java
+++ b/datavec-data/datavec-data-codec/src/main/java/org/datavec/codec/reader/BaseCodecRecordReader.java
@@ -1,0 +1,153 @@
+/*
+ *  * Copyright 2016 Skymind, Inc.
+ *  *
+ *  *    Licensed under the Apache License, Version 2.0 (the "License");
+ *  *    you may not use this file except in compliance with the License.
+ *  *    You may obtain a copy of the License at
+ *  *
+ *  *        http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  *    Unless required by applicable law or agreed to in writing, software
+ *  *    distributed under the License is distributed on an "AS IS" BASIS,
+ *  *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *    See the License for the specific language governing permissions and
+ *  *    limitations under the License.
+ */
+
+package org.datavec.codec.reader;
+
+import org.datavec.api.conf.Configuration;
+import org.datavec.api.records.SequenceRecord;
+import org.datavec.api.records.metadata.RecordMetaData;
+import org.datavec.api.records.metadata.RecordMetaDataURI;
+import org.datavec.api.records.reader.SequenceRecordReader;
+import org.datavec.api.records.reader.impl.FileRecordReader;
+import org.datavec.api.split.InputSplit;
+import org.datavec.api.writable.Writable;
+import org.datavec.image.loader.ImageLoader;
+
+
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Codec record reader for parsing videos
+ *
+ * @author Adam Gibson
+ */
+public abstract class BaseCodecRecordReader extends FileRecordReader implements SequenceRecordReader {
+    protected int startFrame = 0;
+    protected int numFrames = -1;
+    protected int totalFrames = -1;
+    protected double framesPerSecond = -1;
+    protected double videoLength = -1;
+    protected int rows = 28, cols = 28;
+    protected boolean ravel = false;
+
+    public final static String NAME_SPACE = "org.datavec.codec.reader";
+    public final static String ROWS = NAME_SPACE + ".rows";
+    public final static String COLUMNS = NAME_SPACE + ".columns";
+    public final static String START_FRAME = NAME_SPACE + ".startframe";
+    public final static String TOTAL_FRAMES = NAME_SPACE + ".frames";
+    public final static String TIME_SLICE = NAME_SPACE + ".time";
+    public final static String RAVEL = NAME_SPACE + ".ravel";
+    public final static String VIDEO_DURATION = NAME_SPACE + ".duration";
+
+
+    @Override
+    public List<List<Writable>> sequenceRecord() {
+        File next = iter.next();
+
+        try{
+            return loadData(next, null);
+        }catch(IOException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public List<List<Writable>> sequenceRecord(URI uri, DataInputStream dataInputStream) throws IOException {
+        return loadData(null, dataInputStream);
+    }
+
+    protected abstract List<List<Writable>> loadData( File file, InputStream inputStream  ) throws IOException;
+
+
+    @Override
+    public void initialize(Configuration conf, InputSplit split) throws IOException, InterruptedException {
+        setConf(conf);
+        initialize(split);
+    }
+
+    @Override
+    public List<Writable> next(){
+        throw new UnsupportedOperationException("next() not supported for CodecRecordReader (use: sequenceRecord)");
+    }
+
+    @Override
+    public List<Writable> record(URI uri, DataInputStream dataInputStream) throws IOException {
+        throw new UnsupportedOperationException("record(URI,DataInputStream) not supported for CodecRecordReader");
+    }
+
+    @Override
+    public boolean hasNext() {
+        return iter.hasNext();
+    }
+
+    @Override
+    public void setConf(Configuration conf) {
+        super.setConf(conf);
+        startFrame = conf.getInt(START_FRAME,0);
+        numFrames = conf.getInt(TOTAL_FRAMES,-1);
+        rows = conf.getInt(ROWS,28);
+        cols = conf.getInt(COLUMNS,28);
+        framesPerSecond = conf.getFloat(TIME_SLICE,-1);
+        videoLength = conf.getFloat(VIDEO_DURATION,-1);
+        ravel = conf.getBoolean(RAVEL, false);
+        totalFrames = conf.getInt(TOTAL_FRAMES, -1);
+    }
+
+    @Override
+    public Configuration getConf() {
+        return super.getConf();
+    }
+
+    @Override
+    public SequenceRecord nextSequence() {
+        File next = iter.next();
+
+        List<List<Writable>> list;
+        try{
+            list = loadData(next, null);
+        }catch(IOException e){
+            throw new RuntimeException(e);
+        }
+        return new org.datavec.api.records.impl.SequenceRecord(list, new RecordMetaDataURI(next.toURI(), CodecRecordReader.class));
+    }
+
+    @Override
+    public SequenceRecord loadSequenceFromMetaData(RecordMetaData recordMetaData) throws IOException {
+        return loadSequenceFromMetaData(Collections.singletonList(recordMetaData)).get(0);
+    }
+
+    @Override
+    public List<SequenceRecord> loadSequenceFromMetaData(List<RecordMetaData> recordMetaDatas) throws IOException {
+        List<SequenceRecord> out = new ArrayList<>();
+        for(RecordMetaData meta : recordMetaDatas){
+            File f = new File(meta.getURI());
+
+            List<List<Writable>> list = loadData(f, null);
+            out.add(new org.datavec.api.records.impl.SequenceRecord(list, meta));
+        }
+
+        return out;
+    }
+
+
+}

--- a/datavec-data/datavec-data-codec/src/main/java/org/datavec/codec/reader/CodecRecordReader.java
+++ b/datavec-data/datavec-data-codec/src/main/java/org/datavec/codec/reader/CodecRecordReader.java
@@ -18,12 +18,6 @@ package org.datavec.codec.reader;
 
 import org.apache.commons.compress.utils.IOUtils;
 import org.datavec.api.conf.Configuration;
-import org.datavec.api.records.SequenceRecord;
-import org.datavec.api.records.metadata.RecordMetaData;
-import org.datavec.api.records.metadata.RecordMetaDataURI;
-import org.datavec.api.records.reader.SequenceRecordReader;
-import org.datavec.api.records.reader.impl.FileRecordReader;
-import org.datavec.api.split.InputSplit;
 import org.datavec.api.writable.Writable;
 import org.datavec.common.RecordConverter;
 import org.datavec.image.loader.ImageLoader;
@@ -35,14 +29,12 @@ import org.jcodec.common.SeekableByteChannel;
 
 
 import java.awt.image.BufferedImage;
-import java.io.DataInputStream;
 import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
 import java.lang.reflect.Field;
-import java.net.URI;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
 
 /**
@@ -62,47 +54,29 @@ import java.util.List;
  *
  * @author Adam Gibson
  */
-public class CodecRecordReader extends FileRecordReader implements SequenceRecordReader {
-    private int startFrame = 0;
-    private int numFrames = -1;
-    private int totalFrames = -1;
-    private double framesPerSecond = -1;
-    private double videoLength = -1;
+public class CodecRecordReader extends BaseCodecRecordReader {
+
     private ImageLoader imageLoader;
-    private boolean ravel = false;
-
-    public final static String NAME_SPACE = "org.datavec.codec.reader";
-    public final static String ROWS = NAME_SPACE + ".rows";
-    public final static String COLUMNS = NAME_SPACE + ".columns";
-    public final static String START_FRAME = NAME_SPACE + ".startframe";
-    public final static String TOTAL_FRAMES = NAME_SPACE + ".frames";
-    public final static String TIME_SLICE = NAME_SPACE + ".time";
-    public final static String RAVEL = NAME_SPACE + ".ravel";
-    public final static String VIDEO_DURATION = NAME_SPACE + ".duration";
-
 
     @Override
-    public List<List<Writable>> sequenceRecord() {
-        File next = iter.next();
+    public void setConf(Configuration conf) {
+        super.setConf(conf);
+        imageLoader = new ImageLoader(rows,cols);
+    }
 
-        try{
-            return loadData(NIOUtils.readableFileChannel(next));
-        }catch(IOException e){
-            throw new RuntimeException(e);
+    @Override
+    protected List<List<Writable>> loadData( File file, InputStream inputStream  ) throws IOException {
+        SeekableByteChannel seekableByteChannel;
+        if (inputStream != null) {
+            //Reading video from DataInputStream: Need data from this stream in a SeekableByteChannel
+            //Approach used here: load entire video into memory -> ByteBufferSeekableByteChanel
+            byte[] data = IOUtils.toByteArray(inputStream);
+            ByteBuffer bb = ByteBuffer.wrap(data);
+            seekableByteChannel = new FixedByteBufferSeekableByteChannel(bb);
+        } else {
+            seekableByteChannel = NIOUtils.readableFileChannel(file);
         }
-    }
 
-    @Override
-    public List<List<Writable>> sequenceRecord(URI uri, DataInputStream dataInputStream) throws IOException {
-        //Reading video from DataInputStream: Need data from this stream in a SeekableByteChannel
-        //Approach used here: load entire video into memory -> ByteBufferSeekableByteChanel
-        byte[] data = IOUtils.toByteArray(dataInputStream);
-        ByteBuffer bb = ByteBuffer.wrap(data);
-        SeekableByteChannel sbc = new FixedByteBufferSeekableByteChannel(bb);
-        return loadData(sbc);
-    }
-
-    private List<List<Writable>> loadData( SeekableByteChannel seekableByteChannel ) throws IOException {
         List<List<Writable>> record = new ArrayList<>();
 
         if(numFrames >= 1) {
@@ -149,79 +123,6 @@ public class CodecRecordReader extends FileRecordReader implements SequenceRecor
 
         return record;
     }
-
-
-    @Override
-    public void initialize(Configuration conf, InputSplit split) throws IOException, InterruptedException {
-        setConf(conf);
-        initialize(split);
-    }
-
-    @Override
-    public List<Writable> next(){
-        throw new UnsupportedOperationException("next() not supported for CodecRecordReader (use: sequenceRecord)");
-    }
-
-    @Override
-    public List<Writable> record(URI uri, DataInputStream dataInputStream) throws IOException {
-        throw new UnsupportedOperationException("record(URI,DataInputStream) not supported for CodecRecordReader");
-    }
-
-    @Override
-    public boolean hasNext() {
-        return iter.hasNext();
-    }
-
-    @Override
-    public void setConf(Configuration conf) {
-        super.setConf(conf);
-        startFrame = conf.getInt(START_FRAME,0);
-        numFrames = conf.getInt(TOTAL_FRAMES,-1);
-        int rows = conf.getInt(ROWS,28);
-        int cols = conf.getInt(COLUMNS,28);
-        imageLoader = new ImageLoader(rows,cols);
-        framesPerSecond = conf.getFloat(TIME_SLICE,-1);
-        videoLength = conf.getFloat(VIDEO_DURATION,-1);
-        ravel = conf.getBoolean(RAVEL, false);
-        totalFrames = conf.getInt(TOTAL_FRAMES, -1);
-    }
-
-    @Override
-    public Configuration getConf() {
-        return super.getConf();
-    }
-
-    @Override
-    public SequenceRecord nextSequence() {
-        File next = iter.next();
-
-        List<List<Writable>> list;
-        try{
-            list = loadData(NIOUtils.readableFileChannel(next));
-        }catch(IOException e){
-            throw new RuntimeException(e);
-        }
-        return new org.datavec.api.records.impl.SequenceRecord(list, new RecordMetaDataURI(next.toURI(), CodecRecordReader.class));
-    }
-
-    @Override
-    public SequenceRecord loadSequenceFromMetaData(RecordMetaData recordMetaData) throws IOException {
-        return loadSequenceFromMetaData(Collections.singletonList(recordMetaData)).get(0);
-    }
-
-    @Override
-    public List<SequenceRecord> loadSequenceFromMetaData(List<RecordMetaData> recordMetaDatas) throws IOException {
-        List<SequenceRecord> out = new ArrayList<>();
-        for(RecordMetaData meta : recordMetaDatas){
-            File f = new File(meta.getURI());
-
-            List<List<Writable>> list = loadData(NIOUtils.readableFileChannel(f));
-            out.add(new org.datavec.api.records.impl.SequenceRecord(list, meta));
-        }
-
-        return out;
-    }
-
 
     /** Ugly workaround to a bug in JCodec: https://github.com/jcodec/jcodec/issues/24 */
     private static class FixedByteBufferSeekableByteChannel extends ByteBufferSeekableByteChannel {

--- a/datavec-data/datavec-data-codec/src/main/java/org/datavec/codec/reader/NativeCodecRecordReader.java
+++ b/datavec-data/datavec-data-codec/src/main/java/org/datavec/codec/reader/NativeCodecRecordReader.java
@@ -1,0 +1,82 @@
+/*
+ *  * Copyright 2017 Skymind, Inc.
+ *  *
+ *  *    Licensed under the Apache License, Version 2.0 (the "License");
+ *  *    you may not use this file except in compliance with the License.
+ *  *    You may obtain a copy of the License at
+ *  *
+ *  *        http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  *    Unless required by applicable law or agreed to in writing, software
+ *  *    distributed under the License is distributed on an "AS IS" BASIS,
+ *  *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *    See the License for the specific language governing permissions and
+ *  *    limitations under the License.
+ */
+
+package org.datavec.codec.reader;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import org.bytedeco.javacv.FFmpegFrameGrabber;
+import org.bytedeco.javacv.Frame;
+import org.bytedeco.javacv.OpenCVFrameConverter;
+import org.datavec.api.conf.Configuration;
+import org.datavec.api.writable.Writable;
+import org.datavec.common.RecordConverter;
+import org.datavec.image.loader.NativeImageLoader;
+
+/**
+ * An implementation of the CodecRecordReader that uses JavaCV and FFmpeg.
+ *
+ * @author saudet
+ */
+public class NativeCodecRecordReader extends BaseCodecRecordReader {
+
+    private OpenCVFrameConverter.ToMat converter;
+    private NativeImageLoader imageLoader;
+
+    @Override
+    public void setConf(Configuration conf) {
+        super.setConf(conf);
+        converter = new OpenCVFrameConverter.ToMat();
+        imageLoader = new NativeImageLoader(rows,cols);
+    }
+
+    @Override
+    protected List<List<Writable>> loadData( File file, InputStream inputStream ) throws IOException {
+        List<List<Writable>> record = new ArrayList<>();
+
+        try (FFmpegFrameGrabber fg = inputStream != null
+                ? new FFmpegFrameGrabber(inputStream)
+                : new FFmpegFrameGrabber(file)) {
+            if(numFrames >= 1) {
+                fg.start();
+                if(startFrame != 0) fg.setFrameNumber(startFrame);
+
+                for(int i = startFrame; i < startFrame+numFrames; i++) {
+                    Frame grab = fg.grabImage();
+                    record.add(RecordConverter.toRecord(imageLoader.asRowVector(converter.convert(grab))));
+                }
+            } else {
+                if(framesPerSecond < 1)
+                    throw new IllegalStateException("No frames or frame time intervals specified");
+                else {
+                    fg.start();
+
+                    for(double i = 0; i < videoLength; i += framesPerSecond) {
+                        fg.setTimestamp(Math.round(i * 1000000L));
+                        Frame grab = fg.grabImage();
+                        record.add(RecordConverter.toRecord(imageLoader.asRowVector(converter.convert(grab))));
+                    }
+                }
+            }
+        }
+
+        return record;
+    }
+
+}

--- a/datavec-data/datavec-data-codec/src/test/java/org/datavec/codec/reader/CodecReaderTest.java
+++ b/datavec-data/datavec-data-codec/src/test/java/org/datavec/codec/reader/CodecReaderTest.java
@@ -26,6 +26,7 @@ import org.datavec.api.split.FileSplit;
 import org.datavec.api.util.ClassPathResource;
 import org.datavec.api.writable.ArrayWritable;
 import org.datavec.api.writable.Writable;
+import org.junit.Ignore;
 import org.junit.Test;
 
 import java.io.DataInputStream;
@@ -91,7 +92,6 @@ public class CodecReaderTest {
         assertEquals(seqR, fromMeta);
     }
 
-
     @Test
     public void testViaDataInputStream() throws Exception {
 
@@ -121,4 +121,85 @@ public class CodecReaderTest {
         assertEquals(expected,actual);
     }
 
+
+    @Ignore @Test
+    public void testNativeCodecReader() throws Exception {
+        File file = new ClassPathResource("fire_lowres.mp4").getFile();
+        SequenceRecordReader reader = new NativeCodecRecordReader();
+        Configuration conf = new Configuration();
+        conf.set(CodecRecordReader.RAVEL, "true");
+        conf.set(CodecRecordReader.START_FRAME, "160");
+        conf.set(CodecRecordReader.TOTAL_FRAMES, "500");
+        conf.set(CodecRecordReader.ROWS, "80");
+        conf.set(CodecRecordReader.COLUMNS, "46");
+        reader.initialize(new FileSplit(file));
+        reader.setConf(conf);
+        assertTrue(reader.hasNext());
+        List<List<Writable>> record = reader.sequenceRecord();
+//        System.out.println(record.size());
+
+        Iterator<List<Writable>> it = record.iterator();
+        List<Writable> first = it.next();
+//        System.out.println(first);
+
+        //Expected size: 80x46x3
+        assertEquals(1, first.size());
+        assertEquals(80 * 46 * 3, ((ArrayWritable)first.iterator().next()).length());
+    }
+
+    @Ignore @Test
+    public void testNativeCodecReaderMeta() throws Exception {
+        File file = new ClassPathResource("fire_lowres.mp4").getFile();
+        SequenceRecordReader reader = new NativeCodecRecordReader();
+        Configuration conf = new Configuration();
+        conf.set(CodecRecordReader.RAVEL, "true");
+        conf.set(CodecRecordReader.START_FRAME, "160");
+        conf.set(CodecRecordReader.TOTAL_FRAMES, "500");
+        conf.set(CodecRecordReader.ROWS, "80");
+        conf.set(CodecRecordReader.COLUMNS, "46");
+        reader.initialize(new FileSplit(file));
+        reader.setConf(conf);
+        assertTrue(reader.hasNext());
+        List<List<Writable>> record = reader.sequenceRecord();
+        assertEquals(500, record.size());   //500 frames
+
+        reader.reset();
+        SequenceRecord seqR = reader.nextSequence();
+        assertEquals(record, seqR.getSequenceRecord());
+        RecordMetaData meta = seqR.getMetaData();
+//        System.out.println(meta);
+        assertTrue(meta.getURI().toString().endsWith("fire_lowres.mp4"));
+
+        SequenceRecord fromMeta = reader.loadSequenceFromMetaData(meta);
+        assertEquals(seqR, fromMeta);
+    }
+
+    @Ignore @Test
+    public void testNativeViaDataInputStream() throws Exception {
+
+        File file = new ClassPathResource("fire_lowres.mp4").getFile();
+        SequenceRecordReader reader = new NativeCodecRecordReader();
+        Configuration conf = new Configuration();
+        conf.set(CodecRecordReader.RAVEL, "true");
+        conf.set(CodecRecordReader.START_FRAME, "160");
+        conf.set(CodecRecordReader.TOTAL_FRAMES, "500");
+        conf.set(CodecRecordReader.ROWS, "80");
+        conf.set(CodecRecordReader.COLUMNS, "46");
+
+        Configuration conf2 = new Configuration(conf);
+
+        reader.initialize(new FileSplit(file));
+        reader.setConf(conf);
+        assertTrue(reader.hasNext());
+        List<List<Writable>> expected = reader.sequenceRecord();
+
+
+        SequenceRecordReader reader2 = new NativeCodecRecordReader();
+        reader2.setConf(conf2);
+
+        DataInputStream dataInputStream = new DataInputStream(new FileInputStream(file));
+        List<List<Writable>> actual = reader2.sequenceRecord(null, dataInputStream);
+
+        assertEquals(expected,actual);
+    }
 }

--- a/datavec-data/datavec-data-image/pom.xml
+++ b/datavec-data/datavec-data-image/pom.xml
@@ -25,14 +25,6 @@
 
     <artifactId>datavec-data-image</artifactId>
 
-    <properties>
-        <opencv.version>3.1.0</opencv.version>
-        <leptonica.version>1.73</leptonica.version>
-        <javacpp-presets.version>1.3</javacpp-presets.version>
-        <javacpp.version>1.3.1</javacpp.version>
-        <javacv.version>1.3</javacv.version>
-    </properties>
-
     <dependencies>
 
         <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -73,6 +73,12 @@
         <logback.version>1.1.2</logback.version>
         <maven-deploy-plugin.version>2.8.2</maven-deploy-plugin.version>
         <reflections.version>0.9.10</reflections.version>
+        <opencv.version>3.1.0</opencv.version>
+        <ffmpeg.version>3.2.1</ffmpeg.version>
+        <leptonica.version>1.73</leptonica.version>
+        <javacpp-presets.version>1.3</javacpp-presets.version>
+        <javacpp.version>1.3.1</javacpp.version>
+        <javacv.version>1.3.1</javacv.version>
     </properties>
 
     <prerequisites>

